universal-application-tool-0.0.1/test/controllers/dev/DatabaseSeedControllerTest.java

```diff
@@ -0,0 +1,76 @@
+package controllers.dev;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static play.mvc.Http.Status.NOT_FOUND;
+import static play.mvc.Http.Status.OK;
+import static play.test.Helpers.contentAsString;
+import static play.test.Helpers.fakeRequest;
+
+import org.junit.Before;
+import org.junit.Test;
+import play.Mode;
+import play.inject.guice.GuiceApplicationBuilder;
+import play.mvc.Result;
+import repository.WithPostgresContainer;
+
+public class DatabaseSeedControllerTest extends WithPostgresContainer {
+
+  private DatabaseSeedController controller;
+
+  @Before
+  public void setupController() {
+    controller = createControllerInMode(Mode.DEV);
+  }
+
+  @Test
+  public void seedAndClearDatabase_displaysCorrectInformation() {
+    // Navigate to index before seeding - should not have the fake program.
+    Result result = controller.index(fakeRequest().build());
+    assertThat(result.status()).isEqualTo(OK);
+    assertThat(contentAsString(result)).doesNotContain("Mock program");
+
+    // Seed the fake data.
+    result = controller.seed();
+    assertThat(result.status()).isEqualTo(OK);
+    assertThat(contentAsString(result)).contains("The database has been seeded");
+    assertThat(contentAsString(result)).contains("Mock program");
+
+    // Clear the data.
+    result = controller.clear();
+    assertThat(result.status()).isEqualTo(OK);
+    assertThat(contentAsString(result)).contains("The database has been cleared");
+    assertThat(contentAsString(result)).doesNotContain("Mock program");
+  }
+
+  @Test
+  public void index_inNonDevMode_returnsNotFound() {
+    DatabaseSeedController controller = createControllerInMode(Mode.TEST);
+    Result result = controller.index(fakeRequest().build());
+
+    assertThat(result.status()).isEqualTo(NOT_FOUND);
+  }
+
+  @Test
+  public void seed_inNonDevMode_returnsNotFound() {
+    DatabaseSeedController controller = createControllerInMode(Mode.TEST);
+    Result result = controller.seed();
+
+    assertThat(result.status()).isEqualTo(NOT_FOUND);
+  }
+
+  @Test
+  public void clear_inNonDevMode_returnsNotFound() {
+    DatabaseSeedController controller = createControllerInMode(Mode.TEST);
+    Result result = controller.clear();
+
+    assertThat(result.status()).isEqualTo(NOT_FOUND);
+  }
+
+  private DatabaseSeedController createControllerInMode(Mode mode) {
+    return new GuiceApplicationBuilder()
```

Comment:
Question: not sure if this overrides `WithPostgresContainer` causing the test failure. In that case we may need to copy some of the config setup over. Essentially I think evolutions still need to apply.

Reply:
OH! That makes perfect sense to me, thank you for catching that! Will work on making sure the fake is set up correctly

Reply:
You were right! I added our config to a test constants file since it is now used in three places. Using it for the Guice configuration helped these tests pass